### PR TITLE
docs(config): specify aliases and provider priority

### PR DESCRIPTION
**Pull request by anqorithm**
# Summary

Defines the product, technical, and task contract for YAML-configurable model aliases and provider priority.

## Linked Work

- Issue: Relates to #1105
- Spec packet: specs/GH1105/

## Readiness Gate

- [x] Linked issue is in ready_to_spec.
- [x] Product and tech specs are included.
- [x] Change contains no credentials or private deployment details.

## Review Gate

- [x] Agent first-pass review completed.
- [ ] Human final review requested.
- [ ] Maintainer approval recorded.

## Verification

- [x] python3 checks/check_workflow.py --repo .
- [x] python3 checks/check_workflow.py --repo . --spec-dir=specs/GH1105
- [x] cargo fmt --all -- --check
- [x] git diff --check

## Release Notes

- [x] Not user-visible; specification only.

## Agent Disclosure

- [x] Agent assisted; human author review is requested.

Refs #1105

### CODE REVIEW

**Comment by anqorithm** [comment]
SpecRail validation passes and downstream production parity work is waiting on this configuration contract. Please review and merge this spec PR so implementation for #1105 can proceed on the required implementation branch.

**Review by majiayu000 — CHANGES_REQUESTED** [review]
## Summary

Thank you @anqorithm for putting together the GH-1105 specification and for carrying this configuration-parity work forward.

I compared the proposed behavior with upstream Python LiteLLM. It is reasonable for litellm-rs to intentionally provide stricter fail-fast behavior, but the current spec still leaves a few implementation-defining details ambiguous. Please clarify the following in the spec packet:

1. Define `model_aliases` merge behavior as a key-by-key merge, with overlay values winning on duplicate keys; an empty overlay should not clear the base map.
2. Split validation into structural graph validation during config parsing and final target/collision validation after enabled provider models have been expanded, but before health checks or `AppState` publication.
3. Reject alias keys that collide with enabled canonical model names, preserving B-002.
4. Flatten valid transitive aliases to their final canonical target before installation, avoiding runtime hop-limit and declaration-order differences.
5. Reflect these decisions in the corresponding test requirements.

No implementation changes are requested in this PR; these are specification clarifications so the follow-up implementation has one deterministic contract.

## Verdict

Requesting changes narrowly on the specification contract. Once these clarifications are incorporated and the related threads are resolved, please tag me for re-review.

File: specs/GH1105/tech.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+| Config validation | `src/config/validation/config_validators.rs` | Validates providers and router independently | Alias shape and target validation must fail before startup |
+| Router construction | `src/core/router/gateway_config.rs` | Creates deployments, hardcodes priority `0`, installs no aliases | B-001/B-004/B-005/B-007 root behavior |
+| Server startup | `src/server/http.rs` | Builds the unified router from providers and runtime router settings | Must pass validated aliases into construction atomically |
+| Model route | `src/server/routes/ai/models.rs` | Lists deployment model names only | B-002 discoverability |
+| Examples/docs | `config/gateway.yaml.example`, `README.md` | Documents providers and router fields | Migration and operator contract |
+
+## Proposed Design
+
+1. Add `model_aliases: HashMap<String, String>` to `GatewayConfig` with `#[serde(default)]`, include it in defaults/merge/export behavior, and add `priority: u32` to `ProviderConfig` with `#[serde(default)]` and debug/default handling.
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply specs/GH1105/tech.md]: Resolved at exact head `867b9ba0e37cdf3120215a8f70310b7896c136ef`: B-006, tech design #1, T2, and handoff notes now define key-by-key alias-map merge, later-overlay wins per repeated key, omitted/explicit-empty overlays preserve base entries, and no deletion sentinel.

File: specs/GH1105/tech.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+2. Extend gateway validation to reject trimmed-empty aliases/targets, self references, cycles, and chains whose final canonical target is not served by an enabled provider. Use map traversal with a per-alias visited set; do not depend on hash iteration order.
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply specs/GH1105/tech.md]: Resolved at exact head `867b9ba0e37cdf3120215a8f70310b7896c136ef`: validation is split into pure Phase A shape/self/cycle checks and Phase B target/collision validation after enabled provider model expansion, before health checks or publication.

File: specs/GH1105/product.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+- Changing provider-specific outbound model mappings.
+- Adding new routing strategies or changing weight semantics.
+- Adding request-time alias mutation or an alias administration API.
+- Defining automatic failover across different model capabilities.
+
+## Behavior Invariants
+
+1. B-001 A non-empty configured model alias resolves to its configured target for normal and streaming routed requests.
+2. B-002 Configured aliases are included once in the OpenAI-compatible model inventory while canonical deployment models remain available.
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply specs/GH1105/product.md]: Resolved at exact head `867b9ba0e37cdf3120215a8f70310b7896c136ef`: B-002/B-007 and Phase B explicitly reject alias keys that collide with enabled canonical deployment model names, with focused collision coverage assigned to T3.

File: specs/GH1105/product.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+3. B-003 Empty alias names, empty targets, self-aliases, and direct or transitive alias cycles fail configuration/startup explicitly without serving traffic.
+4. B-004 Alias declaration order does not affect resolution, including valid transitive alias chains.
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply specs/GH1105/product.md]: Resolved at exact head `867b9ba0e37cdf3120215a8f70310b7896c136ef`: B-004 and T3 require every valid chain, including chains longer than 16 hops, to be flattened to one alias-to-final-canonical mapping before publication.

File: specs/GH1105/tech.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+| B-004 | deterministic graph validation/installation | reversed-order transitive alias test resolves identically |
+| B-005 | provider config to deployment config | configured tiers select lower priority first |
+| B-006 | serde/defaults and legacy constructor | old YAML parses and creates priority-zero deployments with no aliases |
+| B-007 | validation against enabled-provider models | disabled-only/missing canonical target fails before server state creation |
+| B-008 | deny-unknown schema and pure validation | unknown-field regression and no-I/O unit coverage |
+
+## Data Flow
+
+YAML is environment-substituted and deserialized into `GatewayConfig`. Validation checks the alias graph against models declared by enabled providers. Server startup converts provider entries into runtime deployments, copying priority, then installs aliases into the router snapshot before health checks and `AppState` publication. Routed requests resolve aliases through the existing snapshot path. The model route reads deployment models and configured alias names from the same router state.
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply specs/GH1105/tech.md]: Resolved at exact head `867b9ba0e37cdf3120215a8f70310b7896c136ef`: Phase B derives target validity from staged enabled deployments after configured models, `provider.list_models()`, or the existing fallback expands the canonical model set.

File: specs/GH1105/tech.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (majiayu000) [reply specs/GH1105/tech.md]: Resolved at exact head `867b9ba0e37cdf3120215a8f70310b7896c136ef`: product B-006/edge cases, tech design #1, T2, and tests now define key-wise overlay merge, empty-overlay preservation, and explicitly no clearing sentinel.

**Review by majiayu000 — APPROVED** [review]
所有 5 项规范澄清均已在 `867b9ba` 落实并验证：

1. **Alias 合并语义** — product B-006 / tech design #1 / T2：逐键合并，overlay 同键胜出，省略或显式空 overlay 保留 base，无删除 sentinel。
2. **两阶段校验** — tech design #2：Phase A 在配置解析期做 shape/self/cycle 图校验（不判定目标可用性）；Phase B 在 router 构造期、health check 与 AppState 发布之前做目标与冲突校验。
3. **Canonical 名冲突** — B-002 / B-007 / Phase B：alias key 与已启用 canonical model 同名时拒绝。
4. **传递链扁平化** — B-004：所有合法链（含超过运行时 16-hop 上界的链）在发布前扁平化为单跳映射，声明顺序不影响结果。
5. **测试要求** — T2/T3 与 tech 验收清单已覆盖 key-wise merge、Phase-A 负例、动态展开目标、collision、逆序声明、>16 跳链、原子失败与 priority 路由。

Phase B 从 staged deployment 集合（configured models → `provider.list_models()` → provider-name fallback）派生目标有效性，与 `Router::from_gateway_config` 的实际展开路径一致，闭合了 bot 提出的动态模型缺口。

规范契约已确定，approve。感谢 @anqorithm 的跟进。
